### PR TITLE
dm: remove dependency of secret key from dmctl & remove decrypt command

### PR DESCRIPTION
**Pull request by D3Hunter**
### What problem does this PR solve?

Issue Number: ref #9492

### What is changed and how it works?
- after spec is finalized, we decide to remove dependency of secret key from dmctl, `config export`/`encrypt` sub-commands call api of dm-master
- remove `decrypt` command to avoid user without access to secret key see plaintext password
- some renaming

#### Tests 

 - [x] Unit test
 - [x] Integration test
 - [x] Manual test (add detailed scripts or steps below)
check password in exported config, it's encrypted if secret-key is provided, otherwise plaintext
 - No code

### Release note 

```release-note
Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

If you don't think this PR needs a release note then fill it with `None`.
```

### CODE REVIEW

File: dm/common/config/config.go
Comment (D3Hunter) [inline dm/common/config/config.go:1]: now only dm-master use it, so remove it

File: dm/config/source_config.go [hunk 1]
```diff
@@ -158,20 +158,28 @@ func (c *SourceConfig) Yaml() (string, error) {
...
 }
 
-// Parse parses flag definitions from the argument list.
+// FromToml parses flag definitions from the argument list.
 // accept toml content for legacy use (mainly used by etcd).
-func (c *SourceConfig) Parse(content string) error {
+func (c *SourceConfig) FromToml(content string) error {
 	// Parse first to get config file.
 	metaData, err := toml.Decode(content, c)
-	err2 := c.check(&metaData, err)
```
Comment (D3Hunter) [inline dm/config/source_config.go:166]: `c.check` is inlined

File: dm/config/source_config.go [hunk 2]
```diff
@@ -241,8 +249,6 @@ func (c *SourceConfig) Verify() error {
 		}
 	}
 
-	c.DecryptPassword()
```
Comment (D3Hunter) [inline dm/config/source_config.go:244]: this call does nothing useful, so remove it, below calls too

File: dm/config/task_converters.go [hunk 3]
```diff
@@ -664,8 +664,6 @@ func TaskConfigToOpenAPITask(c *TaskConfig, sourceCfgMap map[string]*SourceConfi
 	cfgs := make(map[string]dbconfig.DBConfig)
 	for _, source := range c.MySQLInstances {
 		if cfg, ok := sourceCfgMap[source.SourceID]; ok {
-			// check the password
-			cfg.DecryptPassword()
```
Comment (D3Hunter) [inline dm/config/task_converters.go:668]: here too, no check is done

File: dm/config/source_config.go [hunk 4]
```diff
@@ -158,20 +158,28 @@ func (c *SourceConfig) Yaml() (string, error) {
...
 	// Parse first to get config file.
 	metaData, err := toml.Decode(content, c)
-	err2 := c.check(&metaData, err)
-	if err2 != nil {
-		return err2
+	if err != nil {
+		return terror.ErrWorkerDecodeConfigFromFile.Delegate(err)
+	}
+	undecoded := metaData.Undecoded()
+	if len(undecoded) > 0 && err == nil {
```
Comment (lance6716) [inline dm/config/source_config.go:170]:
```suggestion
	if len(undecoded) > 0 {
```

seems always err is always nil

**Review by lance6716 — APPROVED** [review]

**Review by GMHDBJD — APPROVED** [review]
LGTM

**Pull request merged**